pandas/core/sorting.py

```diff
@@ -285,6 +316,24 @@ def nargsort(
     return indexer
 
 
+def ensure_key_mapped(values, key):
```

Comment:
A docstring here would be great to have.

Reply:
Added.